src/charm.py

```diff
@@ -110,6 +111,18 @@ def _on_stop(self, event: ops.StopEvent):
     def _on_config_changed(self, event: ops.framework.EventBase) -> None:
         self.configure_charm(event)
 
+    def _set_pool_size_action(self, event: ops.framework.EventBase) -> None:
+        """Set the size for one or more pools."""
+        pools = event.params.get("pools")
+        size = event.params.get("size")
+
+        try:
+            microceph.set_pool_size(pools, size)
+            event.set_results({"status": "success"})
+        except Exception:
```

Comment:
This is mostly a nit but i feel catching base `Exception` class may also prevent exposing other issues (like Timeout exceptions and what not)